### PR TITLE
Adding webhook for travis ci

### PR DESCRIPTION
**Pull request by captncraig**
This adds a webhook for travis ci. It should only accept pushes that result in a successful build.

### CODE REVIEW

**Comment by captncraig** [comment]
With this I wonder if it would be useful to have a `repo.PullCommit(sha)` so that we can update to specifically the commit the webhook is for. that way I can be sure that any commit checked out has passed some level of validation.

**Comment by abiosoft** [comment]
Hi, I made some changes and few refactors.

I added `repo.checkoutCommit(commitHash string)` that can be called after a pull is successful.

Kindly rebase and fix possible conflicts (sorry, if any. :wink: ).

File: travis_hook.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+func (t TravisHook) DoesHandle(h http.Header) bool {
+	return h.Get("Travis-Repo-Slug") != ""
+}
+
+func (t TravisHook) Handle(w http.ResponseWriter, r *http.Request, repo *Repo) (int, error) {
+	if r.Method != "POST" {
+		return http.StatusMethodNotAllowed, errors.New("the request had an invalid method")
+	}
+	if err := t.handleSignature(r, repo.HookSecret); err != nil {
```
Comment (abiosoft) [inline travis_hook.go]: Due to my refactor, this needs to change to `repo.Hook.Secret`.

**Comment by abiosoft** [comment]
Thanks. :+1:

**Pull request merged**